MFA and authorization BE

nagxsan opened the pull request:
Closes #2 

# Summary
- Implemented backend using `bun` and `hono`
- Used `speakeasy` and `qrcode` libraries to generate QR codes and authenticate MFA tokens.
- Used `typeorm` and PostgreSQL as the database.
- Added `rbacMiddleware` as a role based auth middleware for the api routes.

# Ways to start backend
- Duplicate the `.env.example` file as `.env` file in the `packages/server/src` folder and fill in with valid values.
- `cd packages/server` to change to the `server` folder
- `bun run index.ts` to start the backend with `index.ts` as the entrypoint.

ikemHood commented on packages/server/.gitignore:
why remove content of file?

ikemHood commented on packages/server/ormconfig.ts:
Project was setup to use drizzle not typeorm.

ikemHood commented on packages/server/package.json:
```diff
@@ -1,16 +1,24 @@
+    "dev": "bun run --hot index.ts"
   },
   "dependencies": {
-    "drizzle": "^1.4.0",
```
we use drizzle for database accessing and sql. use drizzle over typeorm

nagxsan replied:
Ohh got it, I did `bun init` and then whatever files were generated I simply took that

ikemHood reviewed (commented):
Please, Do not edit .gitignore file. and let's use drizzle to talk to the database

nagxsan commented:
I will push changes for the requests

nagxsan commented:
I have made the requested changes to use drizzle instead of typeorm. Can you please check and let me know if any more changes are needed? Thanks!

nagxsan commented:
Can you please let me know if there are any updates needed from my side? Thanks!

ikemHood commented:
@nagxsan, Please effect changes requested from last review

nagxsan commented:
I did do those changes, I have replaced typeorm with drizzle as well as updated. That's what was needed right? I am sorry but I am not getting what else is to be done.

ikemHood reviewed (commented):
Sorry, made comments without completing the reviews.

nagxsan commented:
I have added a small fix which allows for extensible middlewares from a single template. I do not know the routes and the different access levels so I cannot know before-hand what roles to structure for. But I have added a mechanism using which you can use the `createRbacMiddleware` function to pass in the custom `allowedRoles` list and verify if the current user satisfies one of the supplied roles or not.

Please let me know if this is needed or further changes need to be made.
